packages/snowpack/src/commands/build.ts

```diff
@@ -93,104 +259,67 @@ export async function command(commandOptions: CommandOptions) {
-        logLevel,
+        hmr: false,
+        config: config,
       });
-      allBuiltFromFiles.add(locOnDisk);
-
-      for (const [fileExt, buildResult] of Object.entries(builtFileOutput)) {
-        let {code, map} = buildResult;
-        if (!code) {
-          continue;
-        }
-
-        const outDir = path.dirname(locOnDisk.replace(dirDisk, dirDest));
-        const outFilename = replaceExt(path.basename(locOnDisk), srcExt, fileExt);
-        const outLoc = path.join(outDir, outFilename);
-        const sourceMappingURL = outFilename + '.map';
-
-        switch (fileExt) {
-          case '.css': {
-            if (map) code = cssSourceMappingURL(code, sourceMappingURL);
-            break;
-          }
-          case '.js': {
-            if (builtFileOutput['.css']) {
-              // inject CSS if imported directly
-              const cssFilename = outFilename.replace(/\.js$/i, '.css');
-              code = `import './${cssFilename}';\n` + code;
-            }
-
-            code = wrapImportMeta({code, env: true, isDev: false, hmr: false, config});
-
-            if (map) code = jsSourceMappingURL(code, sourceMappingURL);
-
-            allFilesToResolveImports[outLoc] = {
-              baseExt: fileExt,
-              expandedExt: fileExt,
-              contents: code,
-              locOnDisk,
-            };
-            break;
-          }
-          case '.html': {
-            code = wrapHtmlResponse({code, isDev: false, hmr: false, config});
-            allFilesToResolveImports[outLoc] = {
-              baseExt: fileExt,
-              expandedExt: fileExt,
-              contents: code,
-              locOnDisk,
-            };
-            break;
-          }
-        }
-
-        // make directory if doesn’t exist
-        mkdirp.sync(outDir);
-
-        // write source map
-        if (map) await fs.writeFile(path.join(outDir, sourceMappingURL), map, 'utf-8');
-
-        // write file
-        await fs.writeFile(outLoc, code, getEncodingType(fileExt));
+      await fs.writeFile(importProxyFileLoc, proxyCode, getEncodingType('.js'));
       }
     }
+    return installResult;
+  }
+
+  // 1. Load & build all files for the first time, from source.
+  for (const [fromDisk, dirDest] of mountedDirectories) {
+    const allFiles = glob.sync(`**/*`, {
+      ignore: config.exclude,
+      cwd: fromDisk,
+      absolute: true,
+      nodir: true,
+      dot: true,
+    });
+    for (const rawLocOnDisk of allFiles) {
+      const locOnDisk = path.resolve(rawLocOnDisk); // this is necessary since glob.sync() returns paths with / on windows.  path.resolve() will switch them to the native path separator.
+      const finalDest = locOnDisk.replace(fromDisk, dirDest);
+      const outDir = path.dirname(finalDest);
+      const buildPipelineFile = new FileBuilder(locOnDisk, outDir, buildDirectoryLoc, config);
+      buildPipelineFiles[locOnDisk] = buildPipelineFile;
+      await buildPipelineFile.buildFile();
+    }
```

Comment:
not important, but could anything be gained here from running the file builders in parallel?

Reply:
100%, we'd talked about this a bit but if there are no objections I wanted to implement the feature in one PR and then optimize it in another.

Reply:
Oh yeah agreed; don’t mess with child processes in this PR but my comment was specifically: should we replace the `for (const …` + `await …` with a `Promise.all(…` here?